master/cmd/determined-master/root.go

```diff
@@ -193,18 +193,21 @@ func applyBackwardsCompatibility(configMap map[string]interface{}) (map[string]i
 	vProvisioner, provisionerExisted := configMap["provisioner"]
 
 	// Ensure we use either the old schema or the new one.
-	if (rmExisted || rpsExisted) && (schedulerExisted || provisionerExisted) {
+	oldRMConfig := schedulerExisted || provisionerExisted
+	newRMConfig := rmExisted || rpsExisted
+	if newRMConfig && oldRMConfig {
 		return nil, errors.New(
 			"cannot use the old and the new configuration schema at the same time",
 		)
 	}
-	if rmExisted || rpsExisted {
+	// Use configMap if RMs are not defined at all, or if they are defined using the new schema.
+	if !oldRMConfig {
```

Comment:
I don't know about this change... I think if the resource managers are not defined at all, that it should use the usual path for defaults (e.g., return configMap as is). However, I can see it having bigger impact than intended / unforeseen consequences.